### DISCUSSION TITLE
Bandcamp Search - No API KEY needed

### DISCUSSION THREAD

**Discussion by Tomato6966** [TopiSenpai/LavaSrc#108] [General]
Search Tracks on Bandcamp ( a source lavalink supports for links, but no searches )

Search Endpoint: 
```
https://bandcamp.com/api/nusearch/2/autocomplete?q=<encodedURIComponentQuery>
```
Example:
```
https://bandcamp.com/api/nusearch/2/autocomplete?q=Harry%20Styles%20As%20It%20was
```

It returns:
`{ results: SearchTrack[] }`
and a SearchTrack contains for example:
```json
{
  "type": "t",
  "id": 1171384760,
  "part": "t",
  "art_id": 2920979979,
  "img_id": null,
  "score": -8822,
  "weight": 6.02,
  "bias": 1.0024536,
  "name": "harry_styles_as_it_was",
  "band_id": 4239674296,
  "band_name": "geminis_h1_sectest",
  "album_name": "0x05_sectest_album<b>xD</b>",
  "url": "https://geminish1sectest.bandcamp.com/track/harry-styles-as-it-was",
  "img": "https://f4.bcbits.com/img/a2920979979_3.jpg",
  "album_id": 97856260,
  "stat_params": "search_item_id=1171384760&search_item_type=t&search_match_part=%3F&search_page_id=2794770642&search_page_no=0&search_rank=1&search_sig=9fdeb03c43f51cdd31c571ade2319ec9"
},
```
track url can be passed to lavalink
img is thumbnail
band_name is equal to author
type: "t" is a track

I recommend following headers
```
{
    'User-Agent': 'android-async-http/1.4.1 (http://loopj.com/android-async-http)',
    'Cookie': '$Version=1'
}
```

Also supports api with keys, like that `https://bandcamp.com/api/track/3/info?key=${key}&track_id=279317014` but that's not neat...

**Comment by DRSchlaubi** [comment]
I am not sure how this is related to LavaSrc since BadCamp is a Lavaplayer feature?

**Comment by topi314** [comment]
Does Bandcamp only return tracks?

**Reply by topi314** [reply]
perhaps this should be a lavaplayer feature request instead

**Reply by Tomato6966** [reply]
Bandcamp returns, podcasts, artists, albums, playlists, tracks

**Reply by Tomato6966** [reply]
Maybe it should be a lavaplayer feature request, since it's really simple to implement and could be used in lavaplayer only too

**Reply by topi314** [reply]
then please create an issue with samples & co
